OnlineShop.Services.Order.BusinessLayer/Handlers/DeleteOrderHandler.cs

```diff
@@ -0,0 +1,29 @@
+﻿using AutoMapper;
+using MediatR;
+using OnlineShop.Services.Order.BusinessLayer.Commands;
+using OnlineShop.Services.Order.BusinessLayer.Exceptions;
+using OnlineShop.Services.Order.BusinessLayer.Models.Dto;
+using OnlineShop.Services.Order.DataLayer.Repositories.Interfaces;
+
+namespace OnlineShop.Services.Order.BusinessLayer.Handlers
+{
+    public class DeleteOrderHandler(IOrderRepository orderRepository) : IRequestHandler<DeleteOrderCommand, ResponseDto<object>>
+    {
+        public async Task<ResponseDto<object>> Handle(DeleteOrderCommand request, CancellationToken cancellationToken)
+        {
+            var order = await orderRepository.GetOrderByIdAsync(request.Id);
+
+            if (order is null)
+            {
+                throw new OrderNotFoundException(request.Id);
+            }
+
+            await orderRepository.DeleteOrderAsync(order);
+
+            return new ResponseDto<object>
```

Comment:
return object type is not a good idea. you can create one more model of ResponseDto without Result property, and simply return: 
```
new ResponseDto
            {
                Message = "Successfully deleted."
            };
```
But inherit ` ResponseDto<T>` from ResponseDto as there will be repeated fields